src/napari/layers/labels/labels.py

```diff
@@ -716,35 +718,57 @@ def _get_state(self) -> dict[str, Any]:
     @property
     def selected_label(self):
         """int: Index of selected label."""
-        return self._selected_label
+        # TODO update the implementation by next(reversed(self._selected_labels))
+        # once https://github.com/pyapp-kit/psygnal/pull/395 is accepted
+        # There is no length check here because self._selected_labels
+        # always contains at least one label.
+        return list(self._selected_labels)[-1]
 
     @selected_label.setter
-    def selected_label(self, selected_label):
-        if selected_label == self.selected_label:
+    def selected_label(self, selected_label: int):
+        if (
+            selected_label in self._selected_labels
+            and len(self._selected_labels) == 1
+        ):
             return
         # when setting the label to the background, store the previous
         # otherwise, clear it
-        layer_dtype = get_dtype(self)
-        dtype_lims = get_dtype_limits(layer_dtype)
-        if dtype_lims[0] > selected_label or dtype_lims[1] < selected_label:
-            raise WrongSelectedLabelError(
-                dtype=layer_dtype,
-                value=selected_label,
-                lower_bound=dtype_lims[0],
-                upper_bound=dtype_lims[1],
-            )
         if selected_label == self.colormap.background_value:
             self._prev_selected_label = self.selected_label
         else:
             self._prev_selected_label = None
-        self.colormap.selection = selected_label
-        self._selected_label = selected_label
-        self._selected_color = self.get_color(selected_label)
+        self.selected_labels = [selected_label]
 
         self.events.selected_label()
 
+    @property
+    def selected_labels(self) -> Selection[int]:
+        return self._selected_labels
+
+    @selected_labels.setter
+    def selected_labels(self, selected_labels: Sequence[int]) -> None:
+        if len(selected_labels) == 0:
+            raise ValueError('At least one label must be selected.')
+        layer_dtype = get_dtype(self)
+        dtype_lims = get_dtype_limits(layer_dtype)
+        min_val = min(selected_labels)
+        max_val = max(selected_labels)
+
+        if dtype_lims[0] > min_val or dtype_lims[1] < max_val:
+            raise WrongSelectedLabelError(
+                dtype=layer_dtype,
+                lower_value=min_val,
+                upper_value=max_val,
+                lower_bound=dtype_lims[0],
+                upper_bound=dtype_lims[1],
+            )
+        self._selected_labels.replace_selection(selected_labels)
+        # container when it is changed.
```

Comment:
not sure what this comment means?

Reply:
Oh that was a leftover from an old attempt, sorry! I'll delete this.